plugins/exec.lua

```diff
@@ -11,6 +11,32 @@ local function exec(cmd, keep_newline)
 end
 
 
+local function exec_rw(cmd, keep_newline, str)
+  local proc = process.start { "sh", "-c", cmd }
+  proc:write(str)
+  proc:close_stream(process.STREAM_STDIN)
+  local res = {}
+  local yieldTime=0.01
+  while true do
+    local rdbuf = proc:read_stdout()
+    if not rdbuf then break end
+    if #rdbuf > 0 then table.insert(res, rdbuf) end
+
+    coroutine.yield(yieldTime)
+    if yieldTime < 1 then
+      yieldTime = yieldTime * 2
+    end
+  end
+
+  if proc:returncode() == 127 then
```

Comment:
I guess that this is this true just for `sh`.
If you decide to either make it execute the command directly, or execute the windows shell too, this might not be effective anymore.

Reply:
Sure, the original code of `exec.lua` was (and the other functions still are) bound to *nix / Bash, but if it's planned to boost this plugin to support Windows, then I think it could be done by using a pointer to a table with the needed functions to handle the underlying shells (Bash, PowerShell, cmd, etc.), initialized at load time.

Again, I'm not fluent in Lua so maybe this approach is not the best.

Reply:
You could add a config option with the command to use. And maybe set the default one according to `PLATFORM`.